src/Microsoft.TemplateEngine.Abstractions/Installer/IInstaller.cs

```diff
@@ -33,32 +28,36 @@ public interface IInstaller
         /// Gets latest versions for <paramref name="templatePackages"/>.
         /// </summary>
         /// <param name="templatePackages">the template packages to get latest versions for.</param>
-        /// <returns>list of <see cref="CheckUpdateResult"/> containing latest versions for the sources.</returns>
+        /// <param name="provider"><see cref="IManagedTemplatePackageProvider"/> requesting latest version.</param>
         /// <param name="cancellationToken"></param>
-        Task<IReadOnlyList<CheckUpdateResult>> GetLatestVersionAsync(IEnumerable<IManagedTemplatePackage> templatePackages, CancellationToken cancellationToken);
+        /// <returns>list of <see cref="CheckUpdateResult"/> containing latest versions for the sources.</returns>
+        Task<IReadOnlyList<CheckUpdateResult>> GetLatestVersionAsync(IEnumerable<IManagedTemplatePackage> templatePackages, IManagedTemplatePackageProvider provider, CancellationToken cancellationToken);
 
         /// <summary>
         /// Installs the template package defined by <paramref name="installRequest"/>.
         /// </summary>
         /// <param name="installRequest">the template package to be installed.</param>
-        /// <returns><see cref="InstallResult"/> containing installation results and installed <see cref="InstallResult.TemplatePackage"/> if installation was successful.</returns>
+        /// <param name="provider"><see cref="IManagedTemplatePackageProvider"/> installing the package.</param>
```

Comment:
@DavidKarlas I've added provider to all methods here, though it used only in Install and Update. Open to discuss it but I think it makes sense to add it to all methods.

Reply:
Feels a bit weird, but I also understand reasoning...